ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -903,20 +903,24 @@ chainSelection chainSelEnv chainDiffs =
       -> m (Maybe (ValidatedChainDiff (Header blk) (LedgerDB' blk)))
     go []            = return Nothing
     go (candidate:candidates0) = do
-        mTentativeHeader <- setTentativeHeader
-        validateCandidate chainSelEnv candidate >>= \case
+        -- Remove future blocks from the chain, and update the ChainDB state
+        -- that keeps track of those blocks, so we can take them into account
+        -- in the decision to make blocks available for pipelining.
+        candidate' <- futureCheckCandidate chainSelEnv candidate
```

Comment:
Hmm. Unless you've figure out a more thorough argument, my understanding was that we would leave the existing future check as-is, and merely add a check to only enable pipelining when the translation is available and isn't in the future. At the moment, this diff seems to me to be removing/changing the existing future check instead of leaving it as-is.